alea/include/alps/alea/core.hpp

```diff
@@ -37,39 +37,95 @@ template <typename T>
 struct traits;
 
 /**
- * Data sink as a thin wrapper around a continuous array.
+ * Data view as a thin wrapper around a continuous array.
  *
- * Basically collects a pointer continuous array (vector, Eigen array, C
- * array etc.) together with its size.  This is sufficiently generic for the
- * virtual interface of `estimator`.
+ * Basically collects a pointer continuous array together with its size.  Note
+ * that the view does not own the `data` pointer.
  */
 template <typename T>
-class sink
+class view
 {
 public:
     typedef T value_type;
 
 public:
     /** Construct view on nothing */
-    sink() : data_(nullptr), size_(0) { }
+    view() : data_(nullptr), size_(0) { }
 
     /** Construct view on data area with size */
-    sink(T *data, size_t size) : data_(data), size_(size) { }
+    view(T *data, size_t size)
+        : data_(data), size_(size)
+    { }
 
     /** Get pointer to zeroth element */
     T *data() { return data_; }
 
     /** Get pointer to zeroth element */
     const T *data() const { return data_; }
 
-    /** Get size of data space */
+    /** Return size */
     size_t size() const { return size_; }
 
 private:
     T *data_;
     size_t size_;
 };
 
+/**
+ * Data view as a thin wrapper around a continuous multi-dimensional array.
+ *
+ * Basically collects a pointer continuous array in ROW-MAJOR format (vector,
+ * transposed Eigen array, C array etc.) together with its shape.  Note that
+ * the view neither owns the `data` pointer nor the `shape` pointer.  Thus it
+ * amends `view<T>` with shape information.
+ */
+template <typename T>
+class ndview
```

Comment:
I disagree.  The dimension is a compile-time constant whenever it is or supposed to be a constant at compile time (this is important for multiarrays and Eigen tensors, because loop unrolling is crucial in dereferencing).  Here, the situation is different, since shape is essentially metadata to be attached to datasets.